### ISSUE TITLE
Zemismart version 1.4

### ISSUE THREAD

**Issue by davidfrat123** [YooSangBeom/Smartthings#6]
Hello. 
Thanks for changing the code to incorporate the new TS004F model.
When added as the device handler, Smartthings recognises it as a Tuya Switch and you can assign the 4 buttons. 
But Unfortunately they don't do anything when clicked. It's like ST doesn't realise the button was pressed at all. No event is triggered. 
Hope you can help sort this, I'm terrible with code!

**Comment by QKnoliver** [comment]
The same problem here tried with multiple buttons but no luck.

**Comment by lexiismadd** [comment]
Same problem here too. Device recognised but buttons aren't.

**Comment by QKnoliver** [comment]
the code works. (only for one action per button) first you just need to give the buttons a purpose and then it will register every button (again only 1 press).

**Comment by davidfrat123** [comment]
I’m struggling to get it to work. What exactly do you do?
Add the device handler exactly as it is v1.4 (did you change anything?)
I go on ST, find devices nearby.
Find Tuya button.
Assign what each button does.
But it doesn’t do anything still when I press the button.

**Comment by YooSangBeom** [comment]
It works on my TS004F device.

1. Remove device from ST account.
2. Pair the device.

If it doesn't work...

I think I should look at your event log (ide.smartthings.com - log).

**Comment by lexiismadd** [comment]
I'm also unable to get it to work. The device detects ok but doesn't recognise any button presses. Nothing in the logs either, my logs section is empty.

**Comment by QKnoliver** [comment]
> I'm also unable to get it to work. The device detects ok but doesn't recognise any button presses. Nothing in the logs either, my logs section is empty.

Do you use the hub v2 or v3?

For me it works with v3.
Event list is also active but response is only after you put the first action in. 

But for Davidfrat123 it doesn't work for the v2. 
The event list seems to be empty too with the v2.

**Comment by lexiismadd** [comment]
I'm using v3 as well.

**Comment by lexiismadd** [comment]
I also noticed that I have "pressed" as an action for buttons 1 & 3, but pressed and held for 2 & 4. Not sure if this is normal?

**Comment by QKnoliver** [comment]
Hmm that's not really how it is supposed to be. But try to delete the device and learn it back in. Do you have multiple device handlers for the buttons?

**Comment by lexiismadd** [comment]
I deleted the device, deleted the device handler, readded the device handler, then readded the device.
Still same thing. It detects as a tuya button with 4 buttons but none respond. Also, buttons 2 & 4 have both "pressed" and "held" again

**Comment by lexiismadd** [comment]
I even went as far as to manually delete the child buttons in the ide as well and let it readd them but still same results

**Comment by QKnoliver** [comment]
Can you post screenshots of the buttons, the device events in ide and the device info in ide

**Comment by lexiismadd** [comment]
No problems.  Screenshots attached.
[https://ibb.co/VvByC37](https://ibb.co/VvByC37)
[https://ibb.co/52yS9Gv](https://ibb.co/52yS9Gv)
[https://ibb.co/y8x10b5](https://ibb.co/y8x10b5)
[https://ibb.co/Y3B6yd2](https://ibb.co/Y3B6yd2)
[https://ibb.co/Nn3G2rM](https://ibb.co/Nn3G2rM)
[https://ibb.co/h8DbqLv](https://ibb.co/h8DbqLv)
[https://ibb.co/HnnjMrp](https://ibb.co/HnnjMrp)

**Comment by QKnoliver** [comment]
The button seems not to be connected if it was you would get the modelid and manufacturer without that data it can't proces the actions

**Comment by QKnoliver** [comment]
Can you post photo of the button

**Comment by davidfrat123** [comment]
Thank you. Version 1.6 worked for me! 
Oddly, it didn't work when I tried it first time. 2nd time was the charm tho. 
The button isn't always 100% responsive but this may be because my devices are far apart with walls blocking them. Thank you.

**Comment by QKnoliver** [comment]
@YooSangBeom 

@kkossev has decoded the held action for button 3 and 4 maybe you can update code from his work.

https://github.com/kkossev/SmartThings.git

**Comment by lexiismadd** [comment]
@QKnoliver
Here's the photos of the button

![20210607_060230](https://user-images.githubusercontent.com/75229226/120938474-38439480-c756-11eb-8166-0babc60dfc99.jpg)
![20210607_060241](https://user-images.githubusercontent.com/75229226/120938479-3d084880-c756-11eb-9905-759946c10fcb.jpg)

**Comment by QKnoliver** [comment]
And you are using the latest code? V1.6?

**Comment by lexiismadd** [comment]
Yep I am. Even deleted the old code and readded it just to be sure

**Comment by QKnoliver** [comment]
And the button recognizes it's handler automatically or do you add it manually?

**Comment by lexiismadd** [comment]
It seems to recognise it automatically as a "Tuya button", I don't need to set the handler. When I go in, it shows the zemismart button as the handler.

**Comment by lexiismadd** [comment]
My exact process for adding it is (just in case I'm doing something wrong)

- Open smartthings app on my Android phone
- press on the + 
- on the button, I hold down the bottom right button until the lights start flashing green
- back on the app, I press on scan nearby
- after a few seconds, "Tuya button" appears
- in my app, I go back to the devices area
- I go to "Tuya button"
- I can see buttons 1 & 3 only have "pressed" and buttons 2 & 4 have "pressed" and "held" 
- I assign "turn on / off office light" to button 1 "pressed"
- I wait about a minute to allow any update to the hub from the cloud
- I press button 1.
- nothing happens
- I check the "history" in the smartthings app, it's empty.
- I try refreshing the history a few times with a drag down from the top, but still no event appears

**Comment by QKnoliver** [comment]
It's verry strange that it appears as pressed...

**Comment by QKnoliver** [comment]
But otherwise I don't see anything wrong... So I think it could be a bad button...

**Comment by lexiismadd** [comment]
Damn, that would be unlucky of me. However because of the issues I had with it from the start, the seller did give me a refund anyway. But I kinda had my hopes up that it might still work.
Thanks for checking over everything.

**Comment by QKnoliver** [comment]
Just maybe it will be fixed after a few updates but the point that it's such an unordinary make me question this

**Comment by lexiismadd** [comment]
Yep. I'll keep updating the handler each time it's updated anyway to at least try.. 😊

**Comment by QKnoliver** [comment]
And if you buy a new one best is to search for the "works with smartthings version" and ask that they are sure it's TS0044 it's way easier to use

**Comment by lexiismadd** [comment]
Yep. Thank you @QKnoliver for trying and @YooSangBeom for the code

**Comment by hdker** [comment]
Hello, I bought 9 units, they arrived yesterday. I tested it with the 1.6 and it doesn't work with any. recognise it automatically as a "Tuya button" but when I press nothing happens. Thanks, I Hope you can help.

**Comment by hdker** [comment]
Device photos

https://photos.app.goo.gl/D6dCaCBv3mr8zjsJ8
